lib/internal/modules/esm/resolve.js

```diff
@@ -58,7 +66,20 @@ function defaultResolve(specifier, { parentURL } = {}, defaultResolve) {
       throw new ERR_INPUT_TYPE_NOT_ALLOWED();
   }
 
-  let url = moduleWrapResolve(specifier, parentURL);
+  let url;
+  try {
+    url = moduleWrapResolve(specifier, parentURL);
+  } catch (error) {
+    try {
+      const base = new URL(parentURL);
+      if (base.protocol === 'file:') {
+        const found = resolveAsCommmonJS(specifier, base.pathname);
+        error.message +=
+        `, the resolved CommonJS module would have been: ${found}`;
```

Comment:
We shouldn't assume that `, ` can be added to any resolution message.

Perhaps let's turn it into a suggestion:

```suggestion
        error.message += `\nDid you mean to import ${found}?`
```

Ideally we could even be more contextual in checking if its the missing extension / main etc that is the issue, but the above seems like a nice helper.

Reply:
I suppose we shouldn't output the absolute path for the suggestion though.

Perhaps we could check:

1. If the original specifier was a relative specifier, then we can make `found` relative to `parent`
2. If the original specifier was a bare specifier (doesn't start with `./` or `../` or `/` and doesn't parse as a URL) then to make `found` also look like the original specifier so that `import 'pkg/x'` and be suggested as `'pkg/x.js`?

Also I'm just thinking we probably should only have this apply to `MODULE_NOT_FOUND` errors and not all errors since they are quite specific.

If the above seems like too much work I'm find with keeping the original absolute path though.

Reply:
I will try to implement these checks, they are a great improvement. Thanks.